test/unit/spec/MfaVerify_spec.js

```diff
@@ -1467,6 +1476,253 @@ function (Okta,
+            expect(test.form.passCodeErrorField().length).toBe(1);
+            expect(test.form.passCodeErrorField().text()).toBe('The field cannot be left blank');
+            expect(test.form.errorMessage()).toBe('We found some errors. Please review the form and make corrections.');
+          });
+        });
+        itp('calls verifyFactor with given code if verify button is clicked', function () {
+          return setupEmail().then(function (test) {
+            $.ajax.calls.reset();
+            test.setNextResponse(resChallengeEmail);
+            test.form.emailSendCode().click();
+            return tick(test);
+          })
+          .then(function (test) {
+            $.ajax.calls.reset();
+            test.setNextResponse(resSuccess);
+            test.form.setAnswer('123456');
+            test.form.submit();
+            return Expect.waitForSpyCall(test.successSpy);
+          })
+          .then(function () {
+            expect($.ajax.calls.count()).toBe(1);
+            Expect.isJsonPost($.ajax.calls.argsFor(0), {
+              url: 'https://foo.com/api/v1/authn/factors/emailhp9NXcoXu8z2wN0g3/verify',
+              data: {
+                passCode: '123456',
+                stateToken: 'testStateToken'
+              }
+            });
+          });
+        });
+        itp('calls authClient verifyFactor with rememberDevice URL param', function () {
+          return setupEmail().then(function (test) {
+            $.ajax.calls.reset();
+            test.form.setRememberDevice(true);
+            test.setNextResponse(resChallengeEmail);
+            test.form.emailSendCode().click();
+            return tick(test);
+          })
+          .then(function (test) {
+            $.ajax.calls.reset();
+            test.setNextResponse(resSuccess);
+            test.form.setAnswer('123456');
+            test.form.submit();
+            return Expect.waitForSpyCall(test.successSpy);
+          })
+          .then(function () {
+            expect($.ajax.calls.count()).toBe(1);
+            Expect.isJsonPost($.ajax.calls.argsFor(0), {
+              url: 'https://foo.com/api/v1/authn/factors/emailhp9NXcoXu8z2wN0g3/verify?rememberDevice=true',
+              data: {
+                passCode: '123456',
+                stateToken: 'testStateToken'
+              }
+            });
+          });
+        });
+        itp('temporarily disables the send code button before displaying re-send \
+             to avoid exceeding the rate limit', function () {
+          var deferred = Util.mockRateLimiting();
+          return setupEmail().then(function (test) {
+            test.button = test.form.emailSendCode();
+            expect(test.button.trimmedText()).toEqual('Send email');
+            test.setNextResponse(resChallengeEmail);
+            test.form.emailSendCode().click();
+            return tick().then(function () {
+              expect(test.button.trimmedText()).toEqual('Sent');
+              deferred.resolve();
```

Comment:
should we also check that the button is disabled?
```
expect(test.button.attr('class')).toContain('link-button-disabled');
expect(test.button.attr('class').prop('disabled')).toBe(true);
```
before resolving the promise?